### ISSUE TITLE
Full Bootstrap library

### ISSUE THREAD

**Issue by idleberg** [mdbootstrap/mdb-ui-kit#489]
Right now this is only a theme that requires Bootstrap to be included. Since it should a trivial to add to the build process, I was wondering if you would consider the _option_ to include a full Bootstrap library styled with the Material theme. I know I could build this myself using the supplied LESS, but I'm asking with CDNs in mind (nobody likes extra HTTP requests, right?)

**Comment by FezVrasta** [comment]
Sorry I don't get what do you mean.
1. A package that includes even the Bootstrap source  
   or
2. A Bootstrap fork styled with material design

Thanks.

**Comment by idleberg** [comment]
The second (like the Bootswatch themes)

**Comment by FezVrasta** [comment]
I see. I'd like to know what other guys think about it...  
If it means to fork the Bootstrap source and update our fork everytime they update something then I think it will become quite hard to maintain. I had started a similar repository some weeks ago and I didn't liked it so much

@GradyD ?

**Comment by GradyD** [comment]
I agree with @FezVrasta on this one. Only because anytime Bootstrap makes any changes we would have a difficult merge to do. It is also against best practice to modify bootstrap's core files. Which is why they have created a theme the way they did, http://getbootstrap.com/getting-started/#whats-included-precompiled

I think we can easily accomplish item 1 and package Bootstrap with the release so you do not have to worry about getting the latest version or anything like that.

**Comment by idleberg** [comment]
> Only because anytime Bootstrap makes any changes we would have a difficult merge to do. It is also against best practice to modify bootstrap's core files.

What is difficult about that? It's trivial using npm/bower and grunt. Most themes I know do it that way, and in doubt you could still leave the choice to the user.

**Comment by GradyD** [comment]
We could leave the choice to the user but that would require maintaining two code bases, one with BS and one without. The styles that are used in this theme override, extend, and/or create new markup. Not just changing colors and other minor things. Its not like we copied all of BS.less files and edited the core. We cherry picked our styles based upon what BS has already done. If we merged we would have to override and add our styles directly to the core in order to do what you are proposing. Lets say BS decided to completely rewrite the modal.less file. We no longer have to completely rewrite our modal.less because we are just extending BS. So they can make all the updates in the world and we may only have to write a couple of more lines on our end. Switching to this method is also a breaking change so it would require us to either cut over and only maintain that codebase or move slower as we write out two different LESS files.

I understand what you are asking for, it would be 1 file and easier to work with at times. But the downside outweighs, in my opinion, the advantages. Not to mention that as a developer I strive to always follow the best practice, recommended way (there has to be a reason for the devs to recommend doing it a specific way). 

Does this make sense/help?
